### DISCUSSION TITLE
DOM elements inside canvas without React?

### DISCUSSION THREAD

**Discussion by ghost** [konvajs/konva#1749] [Q&A] [answered]
Apparently, the only way to get real DOM elements into Konva is [DOM Portal](https://konvajs.org/docs/react/DOM_Portal.html) and that's part of the React-only Utils package, isn't it?

I'm looking for a way to have real, interactive components on a draggable, pannable canvas.

**Comment by lavrton** [comment]
You can take a look into DOM code, basically it is just a DOM element with absolute position on top of the canvas. You can do the same with pure Konva.
But generally don't recommend that. DOM elements are ok for temporary purposes (like tooltip or editable text). I would avoid using DOM for permanent elements.

**Comment by ghost** [comment]
@lavrton And how do you create complex elements with images, text, links and form elements if not with real DOM elements?

**Reply by lavrton** [reply, marked as answer]
For such a task, Konva may be not a good fit. Konva is a canvas framework. It is not designed for links and forms. You can emulate and manually draw shapes that will look and work as links and forms.
